tests/poll.rs

```diff
@@ -201,6 +204,49 @@ fn registry_behind_arc() {
     handle2.join().unwrap();
 }
 
+#[test]
+fn poll_no_timeout_not_affected_by_deregister() {
+    // Test to verify issue 1205 does not reproduce again
+    init();
```

Comment:
You can use `init_with_poll`, then you don't need create the `poll` and `events` below.

Reply:
done